pkg/s3wrapper/filesystem_xattr_client.go

```diff
@@ -0,0 +1,117 @@
+package s3wrapper
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func NewFilesystemBasedXattrClient(
+	log logrus.FieldLogger,
+	rootDirectory string,
+) *FilesystemBasedXattrClient {
+	return &FilesystemBasedXattrClient{
+		log:           log,
+		rootDirectory: rootDirectory,
+	}
+}
+
+const (
+	filesystemXattrMetaDataDirectoryName = "metadata"
+	delimiter                            = "::"
```

Comment:
I wanted to test this with a real filesystem that doesn't support extended attributes, so I used VFAT. Then I noticed that `:` (and many other punctuation characters) aren't allowed in VFAT file names, so creating the files failed with `Invaild argument`. I doubt any real user will use VFAT, but just for my testing, can we change this to something that will be allowed in VFAT, for example `;`?